hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -199,7 +202,7 @@ private void maybeThrowLastError() throws IOException {
    */
   @Override
   public void flush() throws IOException {
-    if (supportFlush) {
+    if (!disableOutputStreamFlush) {
```

Comment:
`supportFlush && !disableOutputStreamFlush` ?

Reply:
It was intentional to separate FileSystem APIs (HFlush, HSync) controls and OutputStream Flush API config control. So that there is no confusion on config combinations that needs to be adhered to. Please let me know if you see a concern with this.

Reply:
The only concern I have is that existing customer who relies on `supportFlush` to control `void flush() ` will be affected.

Reply:
Based on how Flush() is already a noop in WASB and ADLS Gen1, the recommendation was to remove the functional flow. But incase any customer has taken a dependency on Flush() being part of driver GA version, we expect the new config to highlight that they need to move to Hflush()/Hsync() specifically for flush operation and understand the perf impact on continuing to use Flush API. As supportFlush is almost always expected to be config set as true, Flush() will be effectively controlled by new config.  Will check with you offline too incase I am missing any point of concern.

Reply:
make sense to me